### ISSUE TITLE
Can not delete users using internal integrations token and API

### ISSUE THREAD

**Issue by RedShiba** [getsentry/self-hosted#3151]
### Self-Hosted Version

24.2.0

### CPU Architecture

x86_64

### Docker Version

25.0.0

### Docker Compose Version

2.24.1

### Steps to Reproduce

1. Create internal integration (Settings -> Custom Integrations -> Create New integration -> Internal Integration)
2. Provide the following permissions to your integration: member:admin, member:read, member:write. 
3. Get token.
4. Try to delete member from sentry using sentry API. (https://docs.sentry.io/api/organizations/delete-an-organization-member/)

Maybe I do smth wrong and this custom integration is not intended for this purpose?

### Expected Result

User is deleted.

### Actual Result

Some users can not be deleted via API. I'm getting an error like:  400 Bad Request "{"detail":"You cannot remove a member who has more access than you."}".

But I can successfully delete users from UI using my own account(I'm owner).

**Comment by azaslavsky** [comment]
Are you logged in as a superadmin? Does the account you are attempting to perform the deletions from have more permissions?

The error message is pretty straightforward: you need greater permissions than those of the member you are trying to delete. Either you (as a superadmin) or someone else with sufficient permissions needs to add them to your member, and then deletion should work.

**Comment by RedShiba** [comment]
Yes, I'm superadmin/owner and can delete this account from UI. But in my understanding, custom internal integration is not dependent on my permissions cuz its not my personal integration but like general, isn't it? I assigned certain permissions(member:admin) to custom integration, got token and expect it can delete any member as mentioned in sentry doc.
I'll re-check my and other account    role/permissions in database and come back, but I believe my permissions is higher due to I can delete via UI.

**Comment by hubertdeng123** [comment]
I believe that custom internal integration permissions are separate from the permissions on your own user. You are able to delete a user through the UI given your user permissions, but that is separate from internal integrations token. I'm going to transfer this over to sentry to see if they have more of an idea of why this may be going on.
